### DISCUSSION TITLE
Merge cells generated from a position file

### DISCUSSION THREAD

**Discussion by bibiboom89** [neperfepx/neper#733] [Q&A]
Hi,

First I would like to thank you for all the work you have done the last 10 years on the software.

I'm trying to generate an aggregate using a position file, but where group of cells can belong to a same grain.

I have one file for :
- the cells positions : Position_File.txt [Position_File.txt](https://github.com/neperfepx/neper/files/12340634/Position_File.txt)
- their crystallographic orientations : Orientation_Data.txt [Orientation_Data.txt](https://github.com/neperfepx/neper/files/12340635/Orientation_Data.txt)
- the grain Id to which they belong to : Cell_Grains_Id.txt [Cell_Grains_Id.txt](https://github.com/neperfepx/neper/files/12340636/Cell_Grains_Id.txt)

The command I use right now is as follow :
`neper -T -n 2460 -dim 3 -domain "cube(253.5,189.5,100)" -morphooptiini "coo:file(Position_File.txt),weight:0" -ori "file(Orientation_Data.txt,des="quaternion")" -group "file(Cell_Grains_Id.txt)" -reg 1 -o simulation `
And here's what I get with the following command line :
`neper -V simulation.tess -datacellcol group -print simulation_agg2`
![simulation_agg2](https://github.com/neperfepx/neper/assets/63070505/2133ece2-ec9c-46a4-a82c-0522fd9cdb0b)

Is there a way to merge all the cells belonging to the same group together ?
And attribute the orientation to the group and not to each cell (which would avoid the repetition of orientations in the fille).
I saw there is a _mergecell_ option with the _-transform_ function, but I'm confused on how it works.

Thank you in advance,
Regards,
Remy

[Position_File.txt](https://github.com/neperfepx/neper/files/12340634/Position_File.txt)
[Orientation_Data.txt](https://github.com/neperfepx/neper/files/12340635/Orientation_Data.txt)
[Cell_Grains_Id.txt](https://github.com/neperfepx/neper/files/12340636/Cell_Grains_Id.txt)

**Comment by rquey** [comment]
A command that works is:
```
neper -T -loadtess simulation.tess -transform "mergecell(group==1,group==2,group==3,group==5,group==6,group==7,group==8,group==9,group==10,group==11,group==12,group==13,group==14,group==15,group==16,group==17,group==18,group==19,group==20,group==21,group==22,group==23)" -o d
neper -V d.tess -datacellcol group -print simulation_agg2
```
![simulation_agg2](https://github.com/neperfepx/neper/assets/2659376/43d2b95a-7b14-4219-a726-f692eed2c22d)

`group==4` does not work because it looks like this:
```
neper -V d.tess -showcell "group==4" -datacellcol group -print simulation_agg2
```
![simulation_agg2](https://github.com/neperfepx/neper/assets/2659376/2b2e4e41-b20d-42d1-931e-100241086ce4)

**Reply by bibiboom89** [reply]
Thank you Romain for your answer.
It seems to be what I'm looking for.
Unfortunately, when I try to reproduce what you proposed, neper crashes.
I think it might be an installation issue, but I'm not sure though. 
If that's the case, could you please guide me toward the package I might be missing ?

Thank again for your support,
Remy

[simulation_OLD.zip](https://github.com/neperfepx/neper/files/12562570/simulation_OLD.zip)
Command line :
```
neper -T -loadtess simulation_OLD.tess -transform "mergecell(group==1,group==2,group==3,group==5,group==6,group==7,group==8,group==9,group==10,group==11,group==12,group==13,group==14,group==15,group==16,group==17,group==18,group==19,group==20,group==21,group==22,group==23)" -o simulation_NEW

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.6.1-7
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 1 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/bibiboom/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -loadtess simulation_OLD.tess -transform
         mergecell(group==1,group==2,group==3,group==5,group==6,group==7,group==8,group==9,group==10,group==11,group==12,group==13,group==14,group==15,group==16,group==17,group==18,group==19,group==20,group==21,group==22,group==23)
         -o simulation_NEW
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Importing tessellation...
Info   :     [i] Parsing file `simulation_OLD.tess'...
Info   :     [i] Parsed file `simulation_OLD.tess'.
Info   : Transforming tessellation...
Info   :   - Merging cells...
Info   :     > Merged 1 cells...
Tess_script_OLD.sh: line 5: 2291697 Aborted                 (core dumped) neper -T -loadtess simulation_OLD.tess -transform "mergecell(group==1,group==2,group==3,group==5,group==6,group==7,group==8,group==9,group==10,group==11,group==12,group==13,group==14,group==15,group==16,group==17,group==18,group==19,group==20,group==21,group==22,group==23)" -o simulation_NEW
```

**Reply by rquey** [reply]
Here is a command you can run:
```
neper -V simulation_OLD.tess -loop GROUP 1 1 23 -showcell group==GROUP -showedge "domtype==1||cell_shown" -print GROUP -endloop
```
For group 4, you'll see this:
![4](https://github.com/neperfepx/neper/assets/2659376/16e0506b-5646-4a67-aae9-93f9b00c7c82)
which Neper's merge command cannot take.
